Applications/spire/source/ui/security_widget.cpp

```diff
@@ -38,57 +38,57 @@ void SecurityWidget::set_widget(QWidget* widget) {
   m_widget->installEventFilter(this);
 }
 
-connection SecurityWidget::connect_security_change_signal(
+connection SecurityWidget::connect_change_security_signal(
     const ChangeSecuritySignal::slot_type& slot) const {
-  return m_security_change_signal.connect(slot);
+  return m_change_security_signal.connect(slot);
 }
 
 bool SecurityWidget::eventFilter(QObject* object, QEvent* event) {
-  if(object == window() || object == m_widget) {
+  if(object == m_widget) {
     if(event->type() == QEvent::KeyPress) {
-      auto e = static_cast<QKeyEvent*>(event);
-      if(e->key() == Qt::Key_PageUp) {
-        if(m_current_security != Security()) {
-          auto s = m_securities.push_front(m_current_security);
-          if(s != Security()) {
-            m_current_security = s;
-            m_security_change_signal(s);
-          }
-        }
-        event->accept();
-        return true;
-      } else if(e->key() == Qt::Key_PageDown) {
-        if(m_current_security != Security()) {
-          auto s = m_securities.push_back(m_current_security);
-          if(s != Security()) {
-            m_current_security = s;
-            m_security_change_signal(s);
-          }
-        }
-        event->accept();
-        return true;
+      keyPressEvent(static_cast<QKeyEvent*>(event));
```

Comment:
If the ```event``` was accepted then you need to return ```false``` to prevent the parent widget from processing it.

Reply:
I've made the change but eventFilter() returns true when the event has been handled/shouldn't be passed to the parent.

Reply:
Agood... agood...